client-lite/src/download/download.cpp

```diff
@@ -432,20 +444,135 @@ UINT Download::_MaxNoProgressIntervals() const
 {
     if (_noProgressTimeout == _unsetTimeout)
     {
-        if (!_mccHost.empty() && _mccManager.NoFallback() && HttpAgent::IsClientError(_httpStatusCode))
+        return g_progressTrackerMaxNoProgressIntervals;
+    }
+
+    auto maxNoProgressIntervals = std::round(std::chrono::duration<double>(_noProgressTimeout) / g_progressTrackerCheckInterval);
+    DO_ASSERT(maxNoProgressIntervals >= 1);
+    return static_cast<UINT>(maxNoProgressIntervals);
+}
+
+std::string Download::_UpdateConnectionTypeAndGetUrl(bool retryAfterFailure)
```

Comment:
>_UpdateConnectionTypeAndGetUr [](http://example.com/codeflow?start=22&length=29)

So the logic here is to oscillate between cdn and mcc? The download will never time out?

Reply:
Oscillate, yes. The no progress timeout is still enabled, no changes to _SchedProgressTracking().